src/main/java/xyz/nucleoid/plasmid/game/common/team/TeamManager.java

```diff
@@ -328,7 +329,28 @@ private ActionResult onDamagePlayer(ServerPlayerEntity player, DamageSource sour
             var playerTeam = this.teamFor(player);
             var attackerTeam = this.teamFor(attacker);
 
-            if (playerTeam != null && playerTeam == attackerTeam && !this.getTeamConfig(playerTeam).friendlyFire()) {
+            boolean friendlyFire = this.getTeamConfig(playerTeam).friendlyFire();
+            boolean indirectFriendlyFire = this.getTeamConfig(playerTeam).indirectFriendlyFire();
+            boolean isProtectedFromDamageSource = false;
+
+            if(source.isOf(DamageTypes.EXPLOSION)
+                    || source.isOf(DamageTypes.PLAYER_EXPLOSION)
+                    || source.isOf(DamageTypes.FIREWORKS)
+                    || source.isOf(DamageTypes.FIREBALL)
+                    || source.isOf(DamageTypes.INDIRECT_MAGIC))
+                isProtectedFromDamageSource = !indirectFriendlyFire; //when blocked friendly fire could also save the attacker, as example from an explosion
+
+            if(source.isOf(DamageTypes.PLAYER_ATTACK)
+                    || source.isOf(DamageTypes.ARROW)
+                    || source.isOf(DamageTypes.TRIDENT)
+                    || source.isOf(DamageTypes.MAGIC)
+                    || source.isOf(DamageTypes.FALLING_BLOCK)
+                    || source.isOf(DamageTypes.FALLING_STALACTITE)
+                    || source.isOf(DamageTypes.FALLING_ANVIL))
+                isProtectedFromDamageSource = !friendlyFire; //case where the attack is direct
+
+
+            if (playerTeam != null && playerTeam == attackerTeam && isProtectedFromDamageSource) {
```

Comment:
Hmm, I might not have fully understood all the context - would it make sense to instead have a check like `playerTeam == attackerTeam && attacker != player`? The `attacker` should already track indirect cause, even with things like TNT (at least in 1.19.4)